pkg/virt-controller/services/template.go

```diff
@@ -200,26 +205,60 @@ func getHypervNodeSelectors(vmi *v1.VirtualMachineInstance) (map[string]string,
 	// The following HyperV features don't require support from the host kernel, according to inspection
 	// of the QEMU sources (4.0 - adb3321bfd)
 	// VAPIC, Relaxed, Spinlocks, VendorID
-	// see https://schd.ws/hosted_files/devconfcz2019/cf/vkuznets_enlightening_kvm_devconf2019.pdf
+	// VPIndex, SyNIC: depend on both MSR and capability
+	// IPI, TLBFlush: depend on KVM Capabilities
+	// Runtime, Reset, SyNICTimer, Frequencies, Reenlightenment: depend on KVM MSRs availability
+	// EVMCS: depends on KVM capability, but the only way to know that is enable it, QEMU doesn't do
+	// any check before that, so we leave it out
+	//
+	// see also https://schd.ws/hosted_files/devconfcz2019/cf/vkuznets_enlightening_kvm_devconf2019.pdf
 	// to learn about dependencies between enlightenments
 
 	hyperv := vmi.Spec.Domain.Features.Hyperv // shortcut
-	if isFeatureStateEnabled(hyperv.VPIndex) {
-		nodeSelectors[NFD_KVM_INFO_PREFIX+"vpindex"] = "true"
-	}
-	if isFeatureStateEnabled(hyperv.Runtime) {
-		nodeSelectors[NFD_KVM_INFO_PREFIX+"runtime"] = "true"
-	}
-	if isFeatureStateEnabled(hyperv.Reset) {
-		nodeSelectors[NFD_KVM_INFO_PREFIX+"reset"] = "true"
-	}
-	if isFeatureStateEnabled(hyperv.SyNIC) {
-		nodeSelectors[NFD_KVM_INFO_PREFIX+"synic"] = "true"
-		// TODO: SyNIC depends on vp-index on QEMU level. We should enforce this constraint.
+	hvFeatureLabels := []hvFeatureLabel{
```

Comment:
What would you think about moving this array out of this function and defining it as a var in the package?

Reply:
fine for me, on it